src/compiler/checker.ts

```diff
@@ -4420,12 +4420,26 @@ namespace ts {
             let container = getThisContainer(node, /*includeArrowFunctions*/ false);
             let parent = container && container.parent;
             if (parent && (isClassLike(parent) || parent.kind === SyntaxKind.InterfaceDeclaration)) {
-                if (!(container.flags & NodeFlags.Static)) {
+                if (!(container.flags & NodeFlags.Static) && !isConstructorParameter(node, container)) {

```

Comment:
Instead of the `isConstructorParameter` function I would just modify the existing test:

```typescript
if (!(container.flags & NodeFlags.Static) &&
    (container.kind !== SyntaxKind.Constructor || isNodeDescendentOf(node, (<ConstructorDeclaration>container).body))) {
    ...
}
```

The `isNodeDescendentOf` is currently a local helper in emitter.ts, so you need to move it to `utilities.ts` so it can be shared.